:sparkles:  Added android tests, for search products

ricardorlg-aval opened the pull request.

Commit:
:sparkles:  Added android tests, for search products

Commit:
Update LottieIdlingAnimationResource.kt

🎨 remove comments

jiovanyvillamilAndroid commented on app/src/androidTest/java/com/cristianvillamil/mercadoapp/idlingresources/OkHttp3CustomIdlingResource.kt:
```diff
@@ -0,0 +1,30 @@
+    override fun getName() = resourceName
+
+    override fun isIdleNow(): Boolean {
+        println(dispatcher.runningCalls().size)
```
Is necessary keep this print?

jiovanyvillamilAndroid commented on app/src/androidTest/java/com/cristianvillamil/mercadoapp/matchers/MotionLayoutMatcher.kt:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    override fun describeMismatch(item: Any?, description: Description) {
+        item as MotionLayout
```
item could be null please check the type first -> item is MotionLayout?

jiovanyvillamilAndroid commented on app/src/androidTest/java/com/cristianvillamil/mercadoapp/screens/SearchScreen.kt:
```diff
@@ -0,0 +1,48 @@
+        withId(R.id.inputSearchEditText)
+    }
+
+    val animation =KView{
```
please ident the code

jiovanyvillamilAndroid commented on app/src/main/java/com/cristianvillamil/mercadoapp/network/BaseUrlProvider.kt:
```diff
@@ -0,0 +1,5 @@
+package com.cristianvillamil.mercadoapp.network
+
+object BaseUrlProvider {
+    var baseUrl="https://api.mercadolibre.com/"
```
ident the code

jiovanyvillamilAndroid reviewed (changes requested):
Please check the code identation

Commit:
:sparkles: Added instrmented tests for product details feature

:art: resolve PR comments
:zap: improve performance of tests
:bug: fix a bug that causes flaky tests fue to OkHttp3 idling times

Commit:
:zap: Improve code, using Kotlin Result class for viewmodel results
